src/components/Header/web3/SignWallet/ConfirmModal.tsx

```diff
@@ -0,0 +1,227 @@
+import { MEDIA_WIDTHS } from 'theme'
+import getShortenAddress from 'utils/getShortenAddress'
+
+const Wrapper = styled.div`
+  margin: 0;
+  padding: 24px 24px;
+  width: 100%;
+  display: flex;
+  gap: 20px;
+  flex-direction: column;
+`
+
+const Highlight = styled.span`
+  color: ${({ theme }) => theme.text};
+  font-weight: 500;
+`
+
+const ModalConfirmProfile: React.FC = () => {
+  const theme = useTheme()
+  const isOpen = useSelector((state: AppState) => state.authen.showConfirmProfile)
+  const setConfirm = useSetConfirmChangeProfile()
+
+  const [connectSuccess, setConnectSuccess] = useState(false)
+  const { signIn } = useLogin()
+  const { account } = useActiveWeb3React()
+  const { pendingAuthentication } = useSessionInfo()
+  const { signedAccount } = useSignedAccountInfo()
+  const [, setKeepCurrentProfile] = useIsKeepCurrentProfile()
+  const { getCacheProfile } = useProfileInfo()
+  const navigate = useNavigate()
+  const isMobile = useMedia(`(max-width: ${MEDIA_WIDTHS.upToExtraSmall}px)`)
+
+  useEffect(() => {
+    if (!isOpen)
+      setTimeout(() => {
+        setConnectSuccess(false)
+      }, 300)
+  }, [isOpen])
+
+  const desiredProfile = getCacheProfile(account ?? '', false)
+  const currentProfile = getCacheProfile(signedAccount ?? '', false)
+
+  const desiredAccountExist = !!desiredProfile
+  const desiredProfileName = desiredProfile?.nickname
+  const currentProfileName = currentProfile?.nickname
+
+  const hideModal = () => {
+    setConfirm(false)
+  }
+
+  const onCancel = async () => {
+    const isGuest = !desiredAccountExist
+    await signIn(isGuest ? undefined : account, isGuest)
+    hideModal()
+  }
+
+  const onConfirm = (force?: boolean) => {
+    if (connectSuccess || force) {
+      hideModal()
+      setKeepCurrentProfile()
+    } else setConnectSuccess(true)
+  }
+
+  const onDismiss = () => onConfirm(true)
+  const renderContent = () => {
+    return (
+      <Column gap="12px">
+        <Text as="p" lineHeight={'20px'}>
+          <Trans>
+            You are connected to KyberSwap with wallet <Highlight>{getShortenAddress(account ?? '')}</Highlight> while
+            your current profile <Highlight>{currentProfileName}</Highlight> was created by wallet{' '}
+            <Highlight>{getShortenAddress(signedAccount ?? '')}</Highlight>. Do you wish to keep using
+          </Trans>
+          &nbsp;
+          {currentProfileName ? (
+            <Trans>
+              profile <Highlight>{currentProfileName}</Highlight>
+            </Trans>
+          ) : (
+            <Trans> this profile</Trans>
+          )}
+          &nbsp;
+          <Trans>
+            or switch to{' '}
+            {desiredAccountExist ? (
+              <>
+                profile <Highlight> {desiredProfileName || getShortenAddress(account ?? '')}</Highlight>
+              </>
+            ) : (
+              t`a guest profile`
```

Comment:
does `t` inside <Trans> work?

Reply:
It does work, but trans for this file is so complicate and using wrong way.